### PR TITLE
[MODULAR] Fixes bug where opening Game Preferences and then spawning causing you to be random human. Also SR bodypart prefs refactor

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

For some reason there doesn't seem to be an issue report for this long-standing issue, whatever.

The reason why opening Game Preferences and then spawning made you random was because we added a check when applying the prefs on spawn if the pref `is_accessible()`. Problem is that `is_accessible()` also filters if the pref is applicable to the current window. Since none of the character prefs are used in Game Options, it filtered out all of them. And also tried to apply game prefs to the player.

All the mutant body parts and cursed body parts prefs relied on this clumsy logic to support being toggled, piggybacking off is_available().

Refactors mutant body part options so that `apply_to_human()` itself handles visibility logic, as it should:
- Almost all the toggles had identical functionality. There is now a basetype `/datum/preference/toggle/mutant_toggle`.
  - The only two that were different are the two donator-only slots. Retained functionality.
- The body part choices are now under `/datum/preference/choiced/mutant_choice`.
  - The only one with added functionality was the snout slot. Updated it to support the current loigic.

Updated `/datum/preference/choiced/genital` likewise.

Also updated the emissive eyes setting.

Note that I tested a bunch of settings, but the volume of preferences and combinations of actions to do probably necessitates this as a test merge.

## How This Contributes To The Skyrat Roleplay Experience

Big, long-standing bug.

## Changelog

:cl:
fix: Opening Game Options and then spawning no longer makes you a random human.
/:cl:

### CODE REVIEW

**Review by GoldenAlpharex — APPROVED** [review]
Finally, we can squash this bug.

I wish our usage of is_accessible wasn't so sinful, but... Oh well.

**Comment by Gandalf2k15** [comment]
This is awful btw and is a bandaid to a much larger issue, don't hide the big main bug

**Comment by Tastyfish** [comment]
> This is awful btw and is a bandaid to a much larger issue, don't hide the big main bug

If anything, the main bug is us adding is_accessible() here. I don't know why we did, but I imagine there was a point to it.

**Comment by Tastyfish** [comment]
Oh and bodypart options getting hidden but stuck in their old state should be less of an issue now.

File: modular_skyrat/master_files/code/modules/client/preferences/_preference.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/client/preferences/_preference.dm]:
```suggestion
 * * target - The character this is being applied to.
```

File: modular_skyrat/master_files/code/modules/client/preferences/_preference.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/client/preferences/_preference.dm:157]: Couldn't we use the default `Mutant Colors` here rather than just going to white? Just a thought, would make a bit more sense to me personally, but it's really whatever.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
A very great fix, and I'd be over the moon happy if this actually fixes the problem, because it's _much_ saner code, and will probably be useful in many other places in the code, probably more than we'd expect.

File: modular_skyrat/master_files/code/modules/client/preferences/_preference.dm
Comment (Tastyfish) [reply modular_skyrat/master_files/code/modules/client/preferences/_preference.dm:157]: The code this was replacing is white - I'm guessing it doesn't matter.

**Comment by Tastyfish** [comment]
On hold for (test) merge until #16019 is merged due to conflicting body part changes.

**Review by GoldenAlpharex — APPROVED** [review]
Should hopefully be ready for a test-merge now that there's no longer any conflicts :)

**Comment by glitchbow** [comment]
For some reason on this round (ID: 6699) my synthetic humanoid spawned in as a human. I actually wasn't sure why I couldn't use *beep, *buzz, *dwoop, and other robotic emotes until I looked and saw this as a bugfix, and thought they might be related. Not a random one, it still has her design and FT, but she got to be a real human for just this round. It is probably related to the fact I hit Game Preferences to change ghost types before joining the round.

yellowarrowpam

**Comment by Tastyfish** [comment]
> For some reason on this round (ID: 6699) my synthetic humanoid spawned in as a human. I actually wasn't sure why I couldn't use *beep, *buzz, *dwoop, and other robotic emotes until I looked and saw this as a bugfix, and thought they might be related. Not a random one, it still has her design and FT, but she got to be a real human for just this round. It is probably related to the fact I hit Game Preferences to change ghost types before joining the round.
> 
> yellowarrowpam

Bizarre, since everything else appears to be applied. Applying the species is still vanilla to upstream.

**Comment by GoldenAlpharex** [comment]
We've seen enough from this for me to be satisfied with the point it's at, let's merge it so we can not have to worry about fixing this test-merge again.

**Pull request merged**
